test/src/test/java/hudson/security/HudsonPrivateSecurityRealmTest.java

```diff
@@ -452,6 +453,29 @@ public void controlCharacterAreNoMoreValid_CustomRegex() throws Exception {
             i++;
         }
     }
+
+    @Test
+    public void createAccountSupportsHashedPasswords() throws Exception {
+        HudsonPrivateSecurityRealm securityRealm = new HudsonPrivateSecurityRealm(false, false, null);
+        j.jenkins.setSecurityRealm(securityRealm);
+
+        securityRealm.createAccount("user_unhashed", "password");
+        securityRealm.createAccount("user_hashed", "#jbcrypt:" + BCrypt.hashpw("password", BCrypt.gensalt()));
```

Comment:
Please add the case the users has a password starting with `#jbcrypt:` but not being a hash. That case is not yet covered.

Reply:
Will do as soon as the comment about implementation above is resolved (ie: which approach I take with this).